src/components/FetchComponent.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from "react";
+
+class FetchComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchJSON = this.fetchJSON.bind(this);
+    /* Placeholder table until we load the data */
+    this.state = {
+      data: {
+        identity: {
+          id: null,
+          login: null,
+        },
+        permissions: {
+          roles: null,
+        },
+      },
+    };
+  }
+  /* Fetches json from mocky API */
+  async fetchJSON() {
```

Comment:
This needs to be in a different file and needs to take parameters, improve error handling.
Also take advantage of the Promise's status, for loaders, etc.

Reply:
I've factored `fetchJSON()` out of the `FetchComponent.js` file into a new file called [`fetchJSON.js`](https://github.com/jql6/mentorsNetwork-FrontEnd/blob/FetchAPI/src/functions/fetchJSON.js) in the `functions` folder.

![fetchJSON](https://user-images.githubusercontent.com/46179386/122304863-c4725c00-ceba-11eb-8378-0995e87798fd.png)

It now takes a parameter for the URL and uses the response status. It also throws an error if the response isn't 200 (OK).